modules/services/mpd-mpris.nix

```diff
@@ -0,0 +1,88 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.services.mpd-mpris;
+
+  mpdCfg = config.services.mpd;
+
+  renderArg = name: value:
+    if lib.isBool value && value then
+      "-${name}"
+    else if lib.isInt value then
+      "-${name} ${toString value}"
+    else if lib.isString value then
+      "-${name} ${lib.escapeShellArg value}"
+    else
+      "";
+
+  concatArgs = strings:
+    lib.concatStringsSep " " (lib.filter (s: s != "") strings);
+
+  renderArgs = args: concatArgs (lib.mapAttrsToList renderArg args);
+
+  renderCmd = pkg: args: "${pkg}/bin/mpd-mpris ${renderArgs args}";
+in {
+  meta.maintainers = [ lib.hm.maintainers.olmokramer ];
+
+  options.services.mpd-mpris = {
+    enable = lib.mkEnableOption
+      "mpd-mpris: An implementation of the MPRIS protocol for MPD";
+
+    package = lib.mkPackageOption pkgs "mpd-mpris" { };
+
+    mpd = {
+      host = lib.mkOption {
+        type = with lib.types; nullOr str;
+        default = if mpdCfg.enable then mpdCfg.network.listenAddress else null;
```

Comment:
I think when `services.mpd` is enabled then it is probably best to simply use `localhost` as the listen address. Doing that seems to make mpd-mpris go for the MPD unix socket: https://github.com/natsukagami/mpd-mpris/blob/11fb84eb87a19965c6f1329d8552ca0ac6439f3d/cmd/mpd-mpris/main.go#L32-L43

Reply:
Alternatively, perhaps the explicit options `-network unix -host %t/mpd/socket` would work?

When having `services.mpd.enabled` it should always listen to the unix socket but may not listen on a TCP socket so this should be more reliable.

Reply:
As far as I can see that function is only called if the `-host` flag is left empty. So maybe the default values for `services.mpd-mpris.mpd.{host,config}` should just be `null`, then no `-host` or `-port` flag is generated and the automatic detection will do its thing. I tried this locally and it seems to work fine.